pkg/instancegroups/instancegroups.go

```diff
@@ -383,40 +375,40 @@ func (c *RollingUpdateCluster) drainTerminateAndWait(ctx context.Context, u *clo
 	return nil
 }
 
-func (c *RollingUpdateCluster) maybeValidate(validationTimeout time.Duration, operation string) error {
+func (c *RollingUpdateCluster) maybeValidate(operation string, validateCount int) error {
 	if c.CloudOnly {
 		klog.Warningf("Not validating cluster as cloudonly flag is set.")
 
 	} else {
 		klog.Info("Validating the cluster.")
 
-		if err := c.validateClusterWithDuration(validationTimeout); err != nil {
+		if err := c.validateClusterWithDuration(validateCount); err != nil {
 
 			if c.FailOnValidate {
-				klog.Errorf("Cluster did not validate within %s", validationTimeout)
-				return fmt.Errorf("error validating cluster after %s a node: %v", operation, err)
+				klog.Errorf("Cluster did not validate within %s", c.ValidationTimeout)
+				return fmt.Errorf("error validating cluster%s: %v", operation, err)
 			}
 
-			klog.Warningf("Cluster validation failed after %s instance, proceeding since fail-on-validate is set to false: %v", operation, err)
+			klog.Warningf("Cluster validation failed%s, proceeding since fail-on-validate is set to false: %v", operation, err)
 		}
 	}
 	return nil
 }
 
 // validateClusterWithDuration runs validation.ValidateCluster until either we get positive result or the timeout expires
-func (c *RollingUpdateCluster) validateClusterWithDuration(validationTimeout time.Duration) error {
-	ctx, cancel := context.WithTimeout(context.Background(), validationTimeout)
+func (c *RollingUpdateCluster) validateClusterWithDuration(validateCount int) error {
```

Comment:
should this function be renamed now that it uses a count parameter rather than a duration parameter?

Reply:
What the function does is impose the timeout. I combined it and `tryValidateCluster`